source/lib/JMSLab/builders/build_latex.py

```diff
@@ -44,3 +46,40 @@ def add_call_args(self):
                                   os.path.normpath(self.log_file))
         self.call_args = args
         return None
+     
+    def add_bib_name(self, target):
+        if bool(target):
+            target = misc.make_list_if_string(target)
+            target_file = str(target[0]).split(".")[0]
+        else:
+            target_file = ''
+        self.bib_file = target_file
+        return None
+    
+    def do_call(self):
+        '''
+        Acutally execute the system call attribute.
+        Raise an informative exception on error.
+        '''
+        self.bibtex_executable  = 'bibtex'
+        self.bibtex_system_call = '%s %s' % (self.bibtex_executable, self.bib_file)
+        
+        traceback = ''
+        raise_system_call_exception = False
+        try:
+            subprocess.check_output(self.system_call, shell = True, stderr = subprocess.STDOUT)
+            subprocess.check_output(self.bibtex_system_call, shell = True, stderr = subprocess.STDOUT)
+            subprocess.check_output(self.system_call, shell = True, stderr = subprocess.STDOUT)
+            subprocess.check_output(self.system_call, shell = True, stderr = subprocess.STDOUT)
+        except subprocess.CalledProcessError as ex:
+            traceback = ex.output
+            raise_system_call_exception = True
+
+        self.cleanup()
```

Comment:
It would be good to have a proper cleanup method for latex builder so that auxiliary files are cleaned after (or maybe also before) compiling.

Reply:
Agree; we can define a dictionary of file extensions to clean up after LaTeX/BibTeX have run, and delete any files with that extension if their name corresponds to the name of the root file. We should only keep `.log` and `.blg`, I think?

Reply:
@mcaceresb and @veli-m-andirin, I added a cleanup function in 92242a4 which I cleaned up in 9f5d9e6. It cleans up the .aux file and the .bbl file after compiling and leaves behind the log files. Let me know what you think of these edits.

Reply:
@veli-m-andirin What's your take on which extensions to include? This is what I have in my own local cleanup file (excluding package-specific extensions). I know many of these probably won't be necessary for the template, though. Should we just include "core" LaTeX (plus `bibtex` and `beamer`)?

```
.aux
.bbl
.bcf
.blg
.brf
.bst
.clo
.cls
.drv
.dtx
.fd
.fls
.glg
.glo
.gls
.glsdefs
.idx
.ilg
.ind
.ins
.ist
.lof
.log
.lot
.nav
.out
.snm
.tikz
.toc
.vrb
```

Reply:
@mcaceresb Just to make sure I am on the same page, you are thinking we would add some or all of these to the list started in line 74 of `build_latex.py`, correct?

Reply:
@mcaceresb Ok, I added these other filetypes (except for .blg since its a log file) to the list of extensions being cleaned up in 40649c3.

Reply:
Thanks @mcaceresb @rcalvo12!

@mcaceresb's list looks great to me. I am also fine with @rcalvo12's idea to exclude .blg. In case @mcaceresb agrees with excluding .blg, I think we can resolve this.

Reply:
Per call with @veli-m-andirin, in 8474693 I added the cleanup function before the compilation as well.